util/src/io/service.rs

```diff
@@ -231,8 +231,8 @@ impl<Message> Handler for IoManager<Message> where Message: Send + Clone + Sync
 	fn notify(&mut self, event_loop: &mut EventLoop<Self>, msg: Self::Message) {
 		match msg {
 			IoMessage::Shutdown => {
-				self.workers.clear();
 				event_loop.shutdown();
```

Comment:
Ok, reverting. I thought that because `event_loop` is not shut down workers may still get additional work because `event_loop` is active. My understanding was that swapping those two lines will block any new events from `event_loop` and after that we will wait for workers to process what's pending.